app/reports/admin_report.rb

```diff
@@ -0,0 +1,12 @@
+class AdminReport
+  def initialize(team)
+    @team = team
+    @responses = if @team
+                   Response.joins(user: { teams: :users_teams })
+                           .where(users_teams: { team_id: @team.id })
+                           .distinct
+                 else
+                   Response.all.distinct
```

Comment:
The Response.joins(user: { teams: :users_teams }) joins the tables responses, users, teams, and users_teams. The .where(users_teams: { team_id: @team.id }) narrows the selected responses to those associated with a specific team.

The reason why .distinct might be necessary here is due to the users_teams table, which is a junction table representing a many-to-many relationship between users and teams.

Here is a scenario where duplicate responses might appear without the .distinct:

Suppose a User is part of multiple Teams (team_1 and team_2).
The User makes a Response (let's call it response_1).
Because the user is part of team_1 and team_2, the response_1 will be associated with both teams due to the joins operation.
Now, when you are generating a report for team_1, response_1 will appear once because of the association with team_1.
But since the user is also part of team_2, response_1 will appear again in the results due to the user's association with team_2.
That's why the .distinct is needed - to remove these duplicate responses that might appear in the results due to a user being part of multiple teams.

3.1.2 :006 > Response.joins(user: { teams: :users_teams }).count
  Response Count (8.0ms)  SELECT COUNT(*) FROM "responses" INNER JOIN "users" ON "users"."id" = "responses"."user_id" INNER JOIN "users_teams" ON "users_teams"."user_id" = "users"."id" INNER JOIN "teams" ON "teams"."id" = "users_teams"."team_id" INNER JOIN "users_teams" "users_teams_teams" ON "users_teams_teams"."team_id" = "teams"."id"
 => `8323 `

3.1.2 :007 > Response.joins(user: { teams: :users_teams }).distinct.count
  Response Count (10.9ms)  SELECT COUNT(DISTINCT "responses"."id") FROM "responses" INNER JOIN "users" ON "users"."id" = "responses"."user_id" INNER JOIN "users_teams" ON "users_teams"."user_id" = "users"."id" INNER JOIN "teams" ON "teams"."id" = "users_teams"."team_id" INNER JOIN "users_teams" "users_teams_teams" ON "users_teams_teams"."team_id" = "teams"."id"
 => `65 `

Reply:
@LyubovP you explain why you add .distinct in line 7, but my question was about line 9
`Response.all.distinct`
you don't have any joins or where operators here
Or you expect that Response.all.count and Response.all.distinct.count will return different value. If so it's bad architecture decision because I expect that all record in any table will be uniq.

Reply:
@SBorozenets Thanks for explaining what you meant about Response.all.distinct.  Yes, you're absolutely correct. In the case of Response.all.distinct, there is indeed no need for distinct.